src/browser/telemetry.js

```diff
@@ -383,7 +383,12 @@ Instrumenter.prototype.instrumentNetwork = function() {
         if (self.trackHttpErrors()) {
           metadata.stack = (new Error()).stack;
         }
-        return orig.apply(this, args).then(function (resp) {
+
+        var promise = orig.apply(this, args);
+
+        // Start our handler before returning the promise. This allows resp.clone()
+        // to execute before other handlers touch the response.
+        promise.then(function (resp) {
```

Comment:
__*Comment*__

Minor suggestion here. I don't think there are strong guarantees on the execution order of handlers to a single promise. Meaning in this case I don't think you can guarantee handlerA runs before handlerB:

```
function handlerA(response) { return response }
function handlerB(response) { return response }

var promise = new Promise(){...}

promise.then(handlerA)
promise.then(handlerB)
```

If your first handler returns the same response as the original (which yours does) you can simply return the promise returned by then to guarantee order

```
function handlerA(response) { return response }
function handlerB(response) { return response }

var promise = new Promise(){...}

var newPromise = promise.then(handlerA)
newPromise.then(handlerB)
```

Reply:
I agree this works based on existing browser behavior, not guaranteed. Great suggestion, I'll update the PR.